src/sources/opentelemetry/http.rs

```diff
@@ -317,12 +354,28 @@ fn decode_metrics_body(
     Ok(events)
 }
 
-async fn handle_request(
+fn serialize_response<T: serde::Serialize + prost::Message>(
+    content_type: ContentType,
+    status_code: hyper::StatusCode,
+    payload: T,
+) -> Result<Response, Rejection> {
+    match content_type {
+        ContentType::Json => {
+            let mut resp = warp::reply::json(&payload).into_response();
+            *resp.status_mut() = status_code;
+            Ok(resp)
+        }
+        ContentType::Protobuf => Ok(protobuf(status_code, payload).into_response()),
+    }
+}
+
+async fn handle_request<T: prost::Message + std::default::Default + serde::Serialize>(
     events: Result<Vec<Event>, ErrorMessage>,
     acknowledgements: bool,
     mut out: SourceSender,
     output: &str,
-    resp: impl Message,
+    resp: T,
```

Comment:
not sure if passing T here is useful or whether `handle_request::<ExportLogs...>()` is better. Neither is great as ideally we'd be able to send a `PartialSuccess` if the batch errored, but that requires knowing how many failed